### PR TITLE
feat(arm): add CKV_AZURE_87 to ensure that Azure Defender is set to On for Key Vault

### PR DESCRIPTION
**Pull request by userrut**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    ansible|argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sast|sca|secrets|serverless|terraform|general|graph|terraform_plan|terraform_json
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

Add a new policy to ensure that Azure Defender is set to On for Key Vault

Fixes # (issue)

### Description
*Include a description of what makes it a violation and any relevant external links.*

### Fix
*How does someone fix the issue in code and/or in runtime?*

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] I have added tests that prove my feature, policy, or fix is effective and works
- [X] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/arm/checks/resource/AzureDefenderOnKeyVaults.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+class AzureDefenderOnKeyVaults(BaseResourceCheck):
+    def __init__(self) -> None:
+        name = "Ensure that Azure Defender is set to On for Key Vault"
+        id = "CKV_AZURE_87"
+        supported_resources = ("Microsoft.Security/pricings",)
+        categories = (CheckCategories.GENERAL_SECURITY,)
+        super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resources)
+
+    def scan_resource_conf(self, conf: dict[str, list[Any]]) -> CheckResult:
+        if conf["properties"]["pricingTier"] == "Standard":
```
Comment (ChanochShayner) [inline checkov/arm/checks/resource/AzureDefenderOnKeyVaults.py:18]: we need to check if the fields are in the dict before accessing them.

**Review by ChanochShayner — COMMENTED** [review]
Nice, please fix your mypy issues.

File: checkov/arm/checks/resource/AzureDefenderOnKeyVaults.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        name = "Ensure that Azure Defender is set to On for Key Vault"
+        id = "CKV_AZURE_87"
+        supported_resources = ("Microsoft.Security/pricings",)
+        categories = (CheckCategories.GENERAL_SECURITY,)
+        super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resources)
+
+    def scan_resource_conf(self, conf: Dict[str, Any]) -> CheckResult:
+        properties = conf.get('properties', {})
+        pricing_tier = properties.get('pricingTier')
+        return (
```
Comment (ChanochShayner) [inline checkov/arm/checks/resource/AzureDefenderOnKeyVaults.py:20]: Probably we should also check the name equal to `KeyVaults`
Am I right @tsmithv11?

Comment (tsmithv11) [reply checkov/arm/checks/resource/AzureDefenderOnKeyVaults.py:20]: Yeah, correct

File: checkov/arm/checks/resource/AzureDefenderOnKeyVaults.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    def __init__(self) -> None:
+        name = "Ensure that Azure Defender is set to On for Key Vault"
+        id = "CKV_AZURE_87"
+        supported_resources = ("Microsoft.Security/pricings",)
+        categories = (CheckCategories.GENERAL_SECURITY,)
+        super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resources)
+
+    def scan_resource_conf(self, conf: Dict[str, Any]) -> CheckResult:
+        properties = conf.get('properties', {})
+        pricing_tier = properties.get('pricingTier')
```
Comment (ChanochShayner) [inline checkov/arm/checks/resource/AzureDefenderOnKeyVaults.py:19]:
```suggestion
        pricing_tier = properties.get('pricingTier')
        name = conf.get('name', '')
```

File: checkov/arm/checks/resource/AzureDefenderOnKeyVaults.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        return (
+            CheckResult.PASSED
+            if pricing_tier == "Standard"
```
Comment (ChanochShayner) [inline checkov/arm/checks/resource/AzureDefenderOnKeyVaults.py:22]:
```suggestion
            if pricing_tier == "Standard" and name == 'KeyVaults'
```

File: tests/arm/checks/resource/example_AzureDefenderOnKeyVaults/fail.json [hunk 5]
```diff
@@ -0,0 +1,19 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "resources": [
+    {
+      "type": "Microsoft.Security/pricings",
+      "apiVersion": "2024-01-01",
+      "name": "fail",
```
Comment (ChanochShayner) [inline tests/arm/checks/resource/example_AzureDefenderOnKeyVaults/fail.json:8]:
```suggestion
      "name": "KeyVaults",
```

File: tests/arm/checks/resource/example_AzureDefenderOnKeyVaults/pass.json [hunk 6]
```diff
@@ -0,0 +1,19 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "resources": [
+    {
+      "type": "Microsoft.Security/pricings",
+      "apiVersion": "2024-01-01",
+      "name": "pass",
```
Comment (ChanochShayner) [inline tests/arm/checks/resource/example_AzureDefenderOnKeyVaults/pass.json:8]:
```suggestion
      "name": "KeyVaults",
```

File: tests/arm/checks/resource/test_AzureDefenderOnKeyVaults.py [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+        test_files_dir = Path(__file__).parent / "example_AzureDefenderOnKeyVaults"
+
+        # when
+        report = Runner().run(root_folder=str(test_files_dir), runner_filter=RunnerFilter(checks=[check.id]))
+
+        # then
+        summary = report.get_summary()
+
+        passing_resources = {
+            "Microsoft.Security/pricings.pass",
```
Comment (ChanochShayner) [inline tests/arm/checks/resource/test_AzureDefenderOnKeyVaults.py:20]:
```suggestion
            "Microsoft.Security/pricings.KeyVaults",
```

File: tests/arm/checks/resource/test_AzureDefenderOnKeyVaults.py [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+        }
+        failing_resources = {
+            "Microsoft.Security/pricings.fail",
```
Comment (ChanochShayner) [inline tests/arm/checks/resource/test_AzureDefenderOnKeyVaults.py:23]:
```suggestion
            "Microsoft.Security/pricings.KeyVaults",
```

**Review by ChanochShayner — APPROVED** [review]

**Review by tsmithv11 — APPROVED** [review]

**Pull request merged**